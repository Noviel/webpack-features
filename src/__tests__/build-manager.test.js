import createBuildManager from '../build-manager';

describe('createBuildManager', () => {
  it('should add only client entries to client target', () => {
    const bm = createBuildManager({
      target: 'client',
      production: true
    });

    bm.addClientEntry({
      name: 'client',
      entry: 'src/client.js'
    });
    
    bm.addServerEntry({
      name: 'server',
      entry: 'src/server.js'
    });

    expect(bm.entries())
      .toMatchSnapshot();
  });

  it('should add only server entries to server target', () => {
    const bm = createBuildManager({
      target: 'server',
      production: true
    });

    bm.addClientEntry({
      name: 'app',
      entry: 'src/app.js'
    });

    bm.addServerEntry({
      name: 'server',
      entry: 'src/server.js'
    });

    expect(bm.entries())
      .toMatchSnapshot();
  });

  it('should add only entries specified in BUILD_APPS', () => {
    const bm = createBuildManager({
      BUILD_APPS: 'second,third',
      target: 'server',
      production: true
    });

    bm.addServerEntry({
      name: 'first',
      entry: 'src/first.js'
    });

    bm.addServerEntry({
      name: 'second',
      entry: 'src/second.js'
    });

    bm.addServerEntry({
      name: 'third',
      entry: 'src/third.js'
    });

    expect(bm.entries())
      .toMatchSnapshot();
  });  
});
