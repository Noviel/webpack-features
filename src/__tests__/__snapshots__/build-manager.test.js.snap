// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBuildManager should add only client entries to client target 1`] = `
Object {
  "client": Array [
    "C:\\\\projects\\\\dev\\\\webpack-features\\\\src\\\\client.js",
  ],
}
`;

exports[`createBuildManager should add only entries specified in BUILD_APPS 1`] = `
Object {
  "second": Array [
    "C:\\\\projects\\\\dev\\\\webpack-features\\\\src\\\\second.js",
  ],
  "third": Array [
    "C:\\\\projects\\\\dev\\\\webpack-features\\\\src\\\\third.js",
  ],
}
`;

exports[`createBuildManager should add only server entries to server target 1`] = `
Object {
  "server": Array [
    "C:\\\\projects\\\\dev\\\\webpack-features\\\\src\\\\server.js",
  ],
}
`;
